test-basic/plan-documents.js

```diff
@@ -50,7 +50,88 @@ describe('documents', function() {
       .catch(done);
     });
   });
-  describe('sample by', function() {
+    describe('group-bucket', function() {
+        it('group-bucket-1', function (done) {
+            execPlan(
+                p.fromLiterals([
+
+                    {"r": 1, "c1": "a", "c2": 1, "c3": "m"},
+                    {"r": 2, "c1": "b", "c2": 1},
+                    {"r": 3, "c1": "a", "c3": "n"},
+                    {"r": 4, "c2": 3, "c3": 0},
+                    {"r": 5, "c1": "b"},
+                    {"r": 6, "c2": 5},
+                    {"r": 7, "c3": "p"},
+                    {"r": 8, "c1": "b", "c2": 1, "c3": "q"}
+
+                ])
+                    .groupToArrays(
+                        p.bucketGroup("rBucket", p.col("r"), [2, 4]),
+                        p.count("numRows")
+                    )
+            )
+                .then(function (response) {
+                    const output = getResults(response);
+                    should(output[0]['rBucket']['value'].length).equal(3);
+                    done();
+                })
+                .catch(done);
+        });
+        it('group-bucket-2', function (done) {
+            execPlan(
+                p.fromLiterals([
+
+                    {"r": 1, "c1": "a", "c2": 1, "c3": "m"},
+                    {"r": 2, "c1": "b", "c2": 1},
+                    {"r": 3, "c1": "a", "c3": "n"},
+                    {"r": 4, "c2": 3, "c3": 0},
+                    {"r": 5, "c1": "b"},
+                    {"r": 6, "c2": 5},
+                    {"r": 7, "c3": "p"},
+                    {"r": 8, "c1": "b", "c2": 1, "c3": "q"}
+
+                ])
+                    .groupToArrays(
+                        p.bucketGroup("rBucket", p.col("r"), [0, 7]),
+                        p.count("numRows")
+                    )
+            )
+                .then(function (response) {
+                    const output = getResults(response);
+                    should(output[0]['rBucket']['value'].length).equal(2);
+                    done();
+                })
+                .catch(done);
+        });
+        it('group-bucket-3', function (done) {
+            execPlan(
+                p.fromLiterals([
+
+                    {"r": 1, "c1": "a", "c2": 1, "c3": "m"},
+                    {"r": 2, "c1": "b", "c2": 1},
+                    {"r": 3, "c1": "a", "c3": "n"},
+                    {"r": 4, "c2": 3, "c3": 0},
+                    {"r": 5, "c1": "b"},
+                    {"r": 6, "c2": 5},
+                    {"r": 7, "c3": "p"},
+                    {"r": 8, "c1": "b", "c2": 1, "c3": "q"}
+
+                ])
+                    .groupToArrays(
+                        p.bucketGroup("rBucket", p.col("r"), [0, 8]),
+                        p.count("numRows")
+                    )
+            )
+                .then(function (response) {
+                    const output = getResults(response);
+                    should(output[0]['rBucket']['value'].length).equal(1);
```

Comment:
This test will throw assertion exception, since the number of bucket created will be 2 ([
  { rBucket_bucket: 1, numRows: 7 },
  { rBucket_bucket: 2, numRows: 1 }
] ). The upper limit is not included

Reply:
I tested and the number of buckets is 1 ( for this example, where the boundaries are [0,8].

![image](https://user-images.githubusercontent.com/100639552/157838294-b93601bf-e943-4d6f-b32f-4cb18e2a1f51.png)

Reply:
Hi @AndreiPanti , please upgrade the server to 10.0.9